src/widget.js

```diff
@@ -1,33 +1,30 @@
-const getFunctionTypeParamsString = params => {
-  if (!params) {
-    return "";
-  }
+const getFunctionTypeParamsString = (params = []) => {
   return params
     .reduce((acc, param) => `${acc}${param.name}: any, `, "")
     .slice(0, -2);
 };
 
+const getEventHandlersTypes = eventHandlers => {
+  return Object.keys(eventHandlers).reduce((acc, value) => {
+    return `${acc} ${value}(eventHandler: (event: any) => void );\n`;
+  }, "");
+};
+
 const getMembersTypes = members => {
   return Object.keys(members).reduce((acc, value) => {
     const { kind, params } = members[value];
     return `${acc} ${
-      kind === "member"
-        ? `${value}: any;\n`
-        : `${value}(${getFunctionTypeParamsString(params)});\n`
+      kind === "function"
+        ? `${value}(${getFunctionTypeParamsString(params)});\n`
+        : `${value}: any;\n`
     }`;
   }, "");
 };
 
-const getWidgetTypeDeclarations = ({ manifest }) => {
-  return Object.keys(manifest).reduce((dts, className) => {
-    const { members = [] } = manifest[className];
-    // should it be part of the $w namespace
-    // need to understand which type should we extend, ($w.Node() or $w.Element)
-    // add typings of widget events
-    return `${dts} interface ${className} extends $w.Node { \n${getMembersTypes(
-      members
-    )} }`;
-  }, "");
+const getWidgetTypeDeclarations = ({ className, members, events }) => {
+  return `interface ${className} extends $w.IFrame { \n${getMembersTypes(
```

Comment:
i think you don't need all those "\n" when using backticks. just go down a line.